File: Server/Server.Application/Users/GetCustomers/GetCustomersQuery.cs
using Server.Application.Abstractions.Messaging;

namespace Server.Application.Users.GetClients;

// public sealed record GetCustomersQuery : IQuery<GetCustomersResponse>;

public sealed class GetCustomersQuery : IQuery<GetCustomersResponse>
{
    public int PageSize { get; init; } = 20;
    public string? Cursor { get; init; }
    public string? SortBy { get; init; } = "CreatedOnUtc";
    public string? SortDirection { get; init; } = "DESC";

    // Filtering properties
    public string? NameFilter { get; init; }
    public string? EmailFilter { get; init; }
    public string? CountryFilter { get; init; }
    public string? CityFilter { get; init; }
    public decimal? MinTotalSpent { get; init; }
    public decimal? MaxTotalSpent { get; init; }
    public int? MinTotalOrders { get; init; }
    public int? MaxTotalOrders { get; init; }
    public DateTime? CreatedAfter { get; init; }
    public DateTime? CreatedBefore { get; init; }
}

File: Client/src/entities/users/api/get-customers/get-customers-query.ts
export type GetCustomersQuery = {
	pageSize?: number;
	cursor?: string | null;
	sortBy?: string | null;
	sortDirection?: string | null;

	// Filtering properties
	nameFilter?: string | null;
	emailFilter?: string | null;
	countryFilter?: string | null;
	cityFilter?: string | null;
	minTotalSpent?: number | null;
	maxTotalSpent?: number | null;
	minTotalOrders?: number | null;
	maxTotalOrders?: number | null;
	createdAfter?: string | null;
	createdBefore?: string | null;
};
